libs/core/src/main/java/org/elasticsearch/common/compatibility/RestApiCompatibleVersion.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.common.compatibility;
+
+/**
+ * A enum representing versions which are used by a REST Compatible API.
+ * A CURRENT instance, represents a major Version.CURRENT from server module.
+ *
+ * Only major versions are supported.
+ */
+public enum RestApiCompatibleVersion {
+
+    V_8(8),
+    V_7(7),
+    V_6(6);//used in testing, to prove validation
```

Comment:
as discussed in person, we can remove this (V_6) and the associated test, we can have other ways to ensure elder version are not in use, and this would probably cause more confusion (even with the comment) then would help.

Reply:
agree, remove the instance and the test